packages/experiments/src/components/Microfeedback/Microfeedback.view.tsx

```diff
@@ -0,0 +1,178 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { withSlots } from '../../Foundation';
+import { Stack } from '../../Stack';
+import { IconButton, IButtonStyles, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { IStackStyles } from 'office-ui-fabric-react/lib/Stack';
+import { Callout } from 'office-ui-fabric-react/lib/Callout';
+import { FocusZone, FocusZoneDirection } from 'office-ui-fabric-react/lib/FocusZone';
+import { List } from 'office-ui-fabric-react/lib/List';
+
+import { IMicrofeedbackComponent, IMicrofeedbackProps } from './Microfeedback.types';
+import { getTheme, mergeStyleSets, FontWeights } from 'office-ui-fabric-react/lib/Styling';
+import { IMicrofeedbackState } from './Microfeedback.state';
+
+import { initializeIcons } from '@uifabric/icons';
+
+const theme = getTheme();
+
+const microfeedbackStyles: IStackStyles = {
+  root: [
+    {
+      margin: 8,
+      float: 'right'
+    }
+  ]
+};
+
+const microfeedbackItemStyles: IButtonStyles = {
+  root: [
+    theme.fonts.small,
+    {
+      width: '100%',
+      color: theme.palette.neutralPrimary,
+      fontWeight: FontWeights.light,
+      fontSize: 12
+    }
+  ]
+};
+
+const styles = mergeStyleSets({
+  title: [
+    theme.fonts.medium,
+    {
+      margin: 8,
+      color: theme.palette.neutralPrimary,
+      fontWeight: FontWeights.semibold
+    }
+  ],
+  body: [
+    theme.fonts.small,
+    {
+      margin: 6,
+      color: theme.palette.neutralPrimary,
+      fontWeight: FontWeights.regular
+    }
+  ]
+});
+
+class MicrofeedbackViewComponent extends React.Component<IMicrofeedbackProps, IMicrofeedbackState> {
+  // ref's will be linked to each of the icons for callout placement
+  private dislikeRef = React.createRef<HTMLDivElement>();
+  private likeRef = React.createRef<HTMLDivElement>();
+
+  constructor(props: IMicrofeedbackProps) {
+    super(props);
+
+    initializeIcons();
+
+    this.state = {
+      // initial state of icons is neutral and followup is not visible
+      vote: 0,
+      isFollowupVisible: false
+    };
+  }
+
+  public render() {
+    const likeIcon = this.state.vote > 0 ? 'LikeSolid' : 'Like';
+    const dislikeIcon = this.state.vote < 0 ? 'DislikeSolid' : 'Dislike';
```

Comment:
How will localization work? We don't generally encode default text in components. Instead we expose strings like this as props.

Reply:
These are icon names. They are not user visible strings.